model_api/cpp/models/src/model_base.cpp

```diff
@@ -27,28 +28,30 @@
 #include <utils/ocv_common.hpp>
 #include <utils/slog.hpp>
 
+void ModelBase::load(std::shared_ptr<InferenceAdapter> adapter) {
+    if (!adapter) {
+        inferenceAdapter = std::make_shared<OpenVINOInferenceAdapter>();
+    } else {
+        inferenceAdapter = adapter;
+    }
+
+    auto core = ov::Core();
+    prepareModel(core);
+}
 
-std::shared_ptr<ov::Model> ModelBase::prepareModel(ov::Core& core) {
+void ModelBase::prepareModel(ov::Core& core) {
     // --------------------------- Read IR Generated by ModelOptimizer (.xml and .bin files) ------------
     /** Read model **/
-    slog::info << "Reading model " << modelFileName << slog::endl;
-    std::shared_ptr<ov::Model> model = core.read_model(modelFileName);
+    slog::info << "Reading model " << modelFile << slog::endl;
+    std::shared_ptr<ov::Model> model = core.read_model(modelFile);
```

Comment:
Would it be possible to add a `InferenceAdapter:getOpenVINOModel` (or something like that) method to the InferenceAdapter interface to avoid calling `core.read_model` here?

Reply:
It means that ov::Model object will live for the whole ModelBase object lifetime and lead to the extra RAM consumption during the inference. We will get rid of this when revise preprocessing embedding step.